Source/Core/DolphinQt/GameList/GameListModel.cpp

```diff
@@ -135,9 +135,20 @@ QVariant GameListModel::data(const QModelIndex& index, int role) const
     break;
   case COL_SIZE:
     if (role == Qt::DisplayRole)
-      return QString::fromStdString(UICommon::FormatSize(game.GetFileSize()));
+    {
+      std::string str = UICommon::FormatSize(game.GetFileSize());
+
+      // Add asterisk to size of compressed files.
+      if (game.GetFileSize() != game.GetVolumeSize())
+        str += '*';
+
+      return QString::fromStdString(str);
+    }
     if (role == Qt::InitialSortOrderRole)
       return static_cast<quint64>(game.GetFileSize());
+    if (role == Qt::ToolTipRole && game.GetFileSize() != game.GetVolumeSize())
+      return QString(tr("* Uncompressed size: ") +
+                     QString::fromStdString(UICommon::FormatSize(game.GetVolumeSize())));
```

Comment:
Do we always want to show this? In particular, WBFS does not store the original size anywhere, so `GetVolumeSize` for WBFS images just returns an upper bound for how large a WBFS image can be (which IIRC always is the size of a dual layer Wii disc).

Reply:
I noticed that. Should I only display it for ciso/gcz?

Reply:
CISO doesn't store the original size either. I'd say we can display it for everything other than WBFS or CISO.

It would probably be best if we added a `virtual bool IsDataSizeAccurate` to the BlobReader class so we don't need to hardcode any blob types in the game list.

Reply:
Adding IsDataSizeAccurate is sensible but I'm just going to change the tooltip to a generic "file is compressed" message for all cases.